src/source/Response.c

```diff
@@ -51,6 +50,11 @@ STATUS createCurlResponse(PCurlRequest pCurlRequest, PCurlResponse* ppCurlRespon
     pCurlResponse->lock = pCallbacksProvider->clientCallbacks.createMutexFn(pCallbacksProvider->clientCallbacks.customData, TRUE);
     CHK(pCurlResponse->lock != INVALID_MUTEX_VALUE, STATUS_INVALID_OPERATION);
 
+    pCurlResponse->dataAvailableLock = pCallbacksProvider->clientCallbacks.createMutexFn(pCallbacksProvider->clientCallbacks.customData, TRUE);
+    CHK(pCurlResponse->dataAvailableLock != INVALID_MUTEX_VALUE, STATUS_INVALID_OPERATION);

```

Comment:
Minor: Would recommend STATUS_NOT_ENOUGH_MEMORY instead.

Reply:
Ok, we have about a half dozen places where we do this, so I can change all of them.

Reply:
I don't think so. This is really a minor question.